app/src/main/java/com/azuresamples/msalnativeauthandroidkotlinsampleapp/MFAFragment.kt

```diff
@@ -0,0 +1,236 @@
+package com.azuresamples.msalnativeauthandroidkotlinsampleapp
+
+import android.app.AlertDialog
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import com.azuresamples.msalnativeauthandroidkotlinsampleapp.databinding.FragmentEmailPasswordBinding
+import com.microsoft.identity.common.java.util.StringUtil
+import com.microsoft.identity.nativeauth.AuthMethod
+import com.microsoft.identity.nativeauth.INativeAuthPublicClientApplication
+import com.microsoft.identity.nativeauth.statemachine.errors.GetAccessTokenError
+import com.microsoft.identity.nativeauth.statemachine.errors.GetAccountError
+import com.microsoft.identity.nativeauth.statemachine.errors.MFAError
+import com.microsoft.identity.nativeauth.statemachine.errors.SignInError
+import com.microsoft.identity.nativeauth.statemachine.results.GetAccessTokenResult
+import com.microsoft.identity.nativeauth.statemachine.results.GetAccountResult
+import com.microsoft.identity.nativeauth.statemachine.results.MFARequiredResult
+import com.microsoft.identity.nativeauth.statemachine.results.SignInResult
+import com.microsoft.identity.nativeauth.statemachine.results.SignOutResult
+import com.microsoft.identity.nativeauth.statemachine.states.AccountState
+import com.microsoft.identity.nativeauth.statemachine.states.MFARequiredState
+import com.microsoft.identity.nativeauth.statemachine.states.SignUpCodeRequiredState
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class MFAFragment : Fragment() {
+
+    private lateinit var authClient: INativeAuthPublicClientApplication
+    private var _binding: FragmentEmailPasswordBinding? = null
+    private val binding get() = _binding!!
+
+    companion object {
+        private val TAG = MFAFragment::class.java.simpleName
+        private enum class STATUS { SignedIn, SignedOut }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        _binding = FragmentEmailPasswordBinding.inflate(inflater, container, false)
+        val view = binding.root
+
+        (activity as? AppCompatActivity)?.supportActionBar?.title = getString(R.string.title_email_otp_mfa)
+
+        authClient = AuthClient.getAuthClient()
+
+        init()
+
+        return view
+    }
+
+    override fun onResume() {
+        super.onResume()
+        getStateAndUpdateUI()
```

Comment:
Why do we need to set the state here through fetching the state from the SDK? Users should only land on this screen if they're performing sign in, i.e. they're not signed in. Which is different from the "landing screens" of the app, which users can land on in any state (signed in or signed out).